### DISCUSSION TITLE
How is bit depth represented in zune-jpeg?

### DISCUSSION THREAD

**Discussion by Shnatsel** [etemesi254/zune-image#99] [Q&A]
I'm trying to integrate `zune-jpeg` into `image` and I have a question about color spaces.

`zune-core` has the [`ColorSpace`](https://docs.rs/zune-core/0.2.1/zune_core/colorspace/enum.ColorSpace.html) enum that doesn't specify the bit depth. Meanwhile `image` has a [`ColorType`](https://docs.rs/image/0.24.5/image/enum.ColorType.html) enum that does specify the bit depth - `Rbga8` and `Rgba16` are distinct values.

How can I convert from the zune representation to the image representation when using `zune-jpeg`? Is the bit depth always assumed to be 8 bit per pixel, or is it communicated through some property I didn't discover?

**Comment by etemesi254** [comment]
Bit depth is a long story.

There is another enum called `BitDepth` in core.

https://github.com/etemesi254/zune-image/blob/712360ef064eb07e098a62c0821a49ddbe48faa0/zune-core/src/bit_depth.rs#L12-L53

The only one of importance now is `BitDepth::Eight` and `BitDepth::Sixteen`, but for jpeg since we do not currently have support for lossless images, we only use `BitDepth::Eight`.

The reason I didn't do it like `image` is that when we want to extend it, it becomes an issue, e.g since they now support F32, you need `Rgba32`, `Rgb32` ans so on and so forth, I saw it fit to be separated.
